internal/cli/atlas/deployments/pause.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/deployments/options"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/atlas/setup"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/log"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
+)
+
+type PauseOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	options.DeploymentOpts
+	store  store.ClusterPauser
+	config setup.ProfileReader
+}
+
+var pauseTemplate = "Pausing deployment '{{.Name}}'.\n"
+
+var ErrDeploymentIsNotIDLE = errors.New("deployment state is not IDLE")
+
+func (opts *PauseOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *PauseOpts) Run(ctx context.Context) error {
+	if err := opts.validateAndPrompt(ctx); err != nil {
+		return err
+	}
+
+	if strings.EqualFold(opts.DeploymentType, options.LocalCluster) {
+		return opts.RunLocal(ctx)
+	}
+
+	return opts.RunAtlas()
+}
+
+func (opts *PauseOpts) RunLocal(ctx context.Context) error {
+	localDeployments, err := opts.GetLocalDeploymentsWithContainers(ctx)
+	if err != nil {
+		return err
+	}
+
+	if deployment, ok := localDeployments[opts.DeploymentName]; ok {
+		if err = opts.pauseContainer(ctx, deployment); err != nil {
+			return err
+		}
+
+		return opts.Print(
+			admin.AdvancedClusterDescription{
+				Name: &opts.DeploymentName,
+			})
+	}
+
+	return options.ErrDeploymentNotFound
+}
+
+func (opts *PauseOpts) pauseContainer(ctx context.Context, deployment options.Deployment) error {
+	if deployment.StateName == options.IdleState {
+		if _, err := opts.PodmanClient.StopContainers(ctx, deployment.MongoTContainer.ID); err != nil {
+			return err
+		}
+
+		return opts.PodmanClient.StopMongoD(ctx, deployment.MongoDContainer.ID)
+	}
+
+	if deployment.StateName == options.PausedState || deployment.StateName == options.StoppedState {
+		return nil
+	}
+
+	return ErrDeploymentIsNotIDLE
+}
+
+func (opts *PauseOpts) RunAtlas() error {
+	if !opts.IsCliAuthenticated() {
+		return ErrNotAuthenticated
+	}
+
+	r, err := opts.store.PauseCluster(opts.ConfigProjectID(), opts.DeploymentName)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *PauseOpts) validateAndPrompt(ctx context.Context) error {
+	if opts.DeploymentName == "" {
+		if err := opts.DeploymentOpts.Select(ctx); err != nil {
+			return err
+		}
+	}
+
+	if opts.DeploymentType == "" {
+		if err := opts.PromptDeploymentType("What would you like to stop?"); err != nil {
+			return err
+		}
+	}
```

Comment:
I think the order of these 2 prompts should be reversed (ask first the type, then pick one of them)